src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -15504,13 +15504,13 @@ void Unit::ProcDamageAndSpellFor(bool isVictim, Unit* target, uint32 procFlag, u
         if (GetTypeId() == TYPEID_PLAYER)
         {
             // On melee based hit/miss/resist need update skill (for victim and attacker)
-            if (procExtra & (PROC_EX_NORMAL_HIT | PROC_EX_MISS | PROC_EX_RESIST))
+            if (procExtra & (PROC_EX_NORMAL_HIT | PROC_EX_MISS | PROC_EX_RESIST | PROC_EX_PARRY | PROC_EX_DODGE))
             {
-                if (target->GetTypeId() != TYPEID_PLAYER && !target->IsCritter())
+                if (target->GetTypeId() != TYPEID_PLAYER)
                     ToPlayer()->UpdateCombatSkills(target, attType, isVictim);
             }
             // Update defence if player is victim and parry/dodge/block
-            else if (isVictim && procExtra & (PROC_EX_DODGE | PROC_EX_PARRY | PROC_EX_BLOCK))
+            else if (isVictim && procExtra & (PROC_EX_BLOCK))
```

Comment:
Doesn't that exclude dodge and parry from increasing your character's defense skill? Is this intended? And if it's intended, shouldn't you update the comment, too?

Reply:
> Doesn't that exclude dodge and parry from increasing your character's defense skill? Is this intended? And if it's intended, shouldn't you update the comment, too?
> 
> Or maybe it doesnn't exclude it from increasing defense, because isVictim can be true in the if clause, too. But why not increase weapons skill on target blocks, too (when parry is counted).

@necropola 
I also noticed I needed to update the comments.

For the first part. 
I had the same exact thought - (I was worried if that I changed it like this then there would be limited ways to increase defense skill) But then I considered the fact that we still are getting defense skill ups when we get hit with basic auto attacks. 

This flows into your second part.
isVictim flowing into defense and offense in the first if clause. 
An argument could definitely be made for increasing weapon skill on block also.
The only reason I didn't it because I don't remember it being blizzlike and it doesn't make much sense to me. 

For example: In terms of increasing proficiency with a "Weapon"
When you parry someone you used that weapon in the action. 
When you block someone you only used the shield in the action, the weapon wasn't used at all. 
That's how I thought about it anyway.

Reply:
> When you block someone you only used the shield in the action, the weapon wasn't used at all.

By that argument you would need to exclude **Dodge**, too.

Anyway, do we have any source on how it worked during WotLK. I did a bit of research, but couldn't find a single line about what counted and what not. What also worries me a bit is that the existing code explicitly excludes skill ups from beating on critters (not that this would make much of a difference), but removing this without evidence doesn't feel good.

Don't get me wrong. I'm all in for making weapon and defense skillups less of an annoyance, e. g. I would love the  `AlwaysMaxSkillForLevel` setting to be activated in **ChromieCraft's** `worldserver.conf`. 😼

Reply:
Yeah the explicit no critter skillups is slightly worrisome. 

But if there was a bad bug directly related to it. I feel like someone would have left a comment.